addon/components/gavant-table.ts

```diff
@@ -0,0 +1,532 @@
+     *
+     * @readonly
+     * @type {boolean} canPanLeft
+     * @memberof GavantTableComponent
+     */
+    @computed(
+        'hasHiddenColumns',
+        'allowFixedCols',
+        'columns.firstObject.{id,valuePath}',
+        'nonFixedColumns.firstObject.{id,valuePath}',
+        'firstVisibleColumn.{id,valuePath}'
+    )
+    get canPanLeft(): boolean {
+        //columns can be panned left if there are hidden columns
+        //and the first visible column (first NON-FIXED visible column, if fixed columns are enabled)
+        //is not the first defined column
+        const firstColId = this.getColumnId(get(get(this, 'columns'), 'firstObject'));
+        const nonFixedColId = this.getColumnId(get(get(this, 'nonFixedColumns'), 'firstObject'));
+        const colId = this.allowFixedCols ? nonFixedColId : firstColId;
+        const visibleColId = this.getColumnId(this.firstVisibleColumn);
+        return this.hasHiddenColumns && Boolean(colId && visibleColId) && colId !== visibleColId;
+    }
+
+    /**
+     * CP that returns a unique data-table ID
+     *
+     * @readonly
+     * @type {string} tableId
+     * @memberof GavantTableComponent
+     */
+    @computed('elementId')
+    get tableId(): string {
+        return `data-table-${this.elementId}`;
+    }
+
+    /**
+     * CP that returns a CSS style for the table height
+     *
+     *
+     * @readonly
+     * @type {string} height
+     * @memberof GavantTableComponent
+     */
+    @computed('tableHeight')
+    get height(): SafeString {
+        return htmlSafe(this.tableHeight ? `height: ${this.tableHeight};` : '');
+    }
+
+    /**
+     * CP that returns a class string that includes all of
+     * the enabled properties
+     *
+     * @readonly
+     * @type {string} tableClassNames
+     * @memberof GavantTableComponent
+     */
+    @computed('tableClass', 'showHeader', 'stripedRows', 'hoverableRows', 'clickableRows', 'small')
+    get tableClassNames(): string {
+        const classNames = A([this.tableClass]);
+
+        if (!this.showHeader) {
+            classNames.pushObject('table-no-header');
+        }
+
+        if (this.stripedRows) {
+            classNames.pushObject('table-striped');
+        }
+
+        if (this.hoverableRows) {
+            classNames.pushObject('table-hover');
+        }
+
+        if (this.clickableRows) {
+            classNames.pushObject('table-clickable-rows');
+        }
+
+        if (this.small) {
+            classNames.pushObject('table-sm');
+        }
+
+        return classNames.join(' ');
+    }
+
+    /**
+     * CP that returns a new fillColumnIndex when the current
+     * fillColumnIndex is out of range
+     *
+     * @readonly
+     * @type {(number | null)} adjustedFillColumnIndex
+     * @memberof GavantTableComponent
+     */
+    @computed('fillColumnIndex', 'visibleColumns.length')
+    get adjustedFillColumnIndex(): number | null {
+        if (this.fillColumnIndex) {
```

Comment:
I find this statement sort of confusing. If `this object` else return `this object`. Might not hurt to just return null (if that's what its returning in that case)

Reply:
I am pretty sure we are handling the non-zero case here, so this might be safe to change to `if (this.fillColumnIndex > 0) { ...` to be more descriptive. but i would double-check that @napafundi